ferry.go

```diff
@@ -278,7 +316,12 @@ func (f *Ferry) Start() error {
 	// miss some records that are inserted between the time the
 	// DataIterator determines the range of IDs to copy and the time that
 	// the starting binlog coordinates are determined.
-	err := f.BinlogStreamer.ConnectBinlogStreamerToMysql()
+	var err error
+	if f.StateToResumeFrom != nil {
```

Comment:
i dont know how i feel about all these conditionals on `f.StateToResumeFrom == nil`. can we somehow boil it down to a single statement at a "higher" level?

Reply:
Yeah we can. I'll have to cherry-pick some changes from a downstream branch that I have going on. It's not perfect but it's a bit better.

Reply:
Hmm I can't easily cherry pick it back as I ended up moved the table loading logic from `Start` to `Initialize` and used the `StateTracker` instead of `StateToResumeFrom` directly in `Start` after adding in the Reconciler.

I prefer if we can leave this code as is, as the PRs coming up will change this section and hopefully will make this logic less confusing.